modules/imap/functions.php

```diff
@@ -1662,3 +1666,39 @@ function flattenMessagesLists($messagesLists, $listSize) {
 
     return ['messages' => $endList, 'offsets' => $sizesTaken];
 }
+
+if (!hm_exists('save_sent_msg')) {
+function save_sent_msg($handler, $imap_id, $imap, $imap_details, $msg, $msg_id, $show_errors = true) {
+    $specials = get_special_folders($handler, $imap_id);
+    if (array_key_exists('sent', $specials) && $specials['sent']) {
+        $sent_folder = $specials['sent'];
+    }
+
+    if (!$sent_folder) {
+        $auto_sent = $imap->get_special_use_mailboxes('sent');
+        if (!array_key_exists('sent', $auto_sent)) {
+            return;
+        }
+        $sent_folder = $auto_sent['sent'];
+    }
+    if (!$sent_folder) {
+        Hm_Debug::add(sprintf("Unable to save sent message, no sent folder for IMAP %s", $imap_details['server']));
+    }
+    $uid = null;
+    if ($sent_folder) {
+        Hm_Debug::add(sprintf("Attempting to save sent message for IMAP server %s in folder %s", $imap_details['server'], $sent_folder));
+        if (! $imap->store_message($sent_folder, $msg)) {
```

Comment:
store_message should already return the uid of the new msg - no need to get the list of messages once again to retrieve the uid.